### PR TITLE
`CopyToClipboard` Component

### PR DESCRIPTION
**Pull request by hobbescodes**
##### Task link: https://trello.com/c/xe59l6fQ/248-copy-to-clipboard-button

Added `CopyToClipboard` component, stories, and tests. Updated exported hooks (backfill from `usehooks-ts`).

## Test Steps

1) Verify that component and story structure (and logic) is sound.
2) Verify that component renders appropriately in storybook and in downstream example apps.
3) Verify that tests pass.
4) Verify that build includes all hooks from `usehooks-ts`.

### CODE REVIEW

File: src/components/utility/CopyToClipboard/CopyToClipboard.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export const copyState = async <R extends Renderer = ReactRenderer>({
+  canvasElement,
+  step,
+}: PlayFunctionContext<R>) => {
+  const canvas = within(canvasElement as HTMLElement);
+
+  const user = userEvent.setup();
+
+  const trigger = await canvas.findByRole("button");
+  //   const input = await canvas.findByRole("textbox");
```
[the remark this answers is not in the document]
Comment (hobbescodes) [reply src/components/utility/CopyToClipboard/CopyToClipboard.spec.tsx]: Yeah, I left it there for the test that was commented out. Removed and updated the current testing suite. Currently it only tests that the tooltip will render on hover. Added a TODO to flesh out the interaction tests more in the future when we do more E2E natured stuff and have access to the clipboard API.

**Comment by coopbri** [comment]
Everything works great. One minor style suggestion: I think setting the icon to have `mb={1}` looks better than `mb={1.5}`.

1.5 (current):
mb15

1:
mb1

What do you think?

**Comment by hobbescodes** [comment]
> Everything works great. One minor style suggestion: I think setting the icon to have `mb={1}` looks better than `mb={1.5}`.
> 
> 1.5 (current): mb15
> 
> 1: mb1
> 
> What do you think?

I agree, I would say even better yet might be make the button a flex container, align items center, and add a gap of 1. Then get rid of the nit picky margin stuff on the `Icon` component. Let me know your thoughts!

**Review by coopbri — APPROVED** [review]
Sounds good to me! Thanks for this handy component!

**Pull request merged**